### ISSUE TITLE
Proposal: merge these changes into the official Helm repository

### ISSUE THREAD

**Issue by vielmetti** [jessestuart/tiller-multiarch#1]
This project is great, but it would be also be great if it wasn't needed, and it wouldn't be needed if the official Tiller build had these changes merged in for multiarch.

As requested here: 

* "Release new tiller docker image on arm64", https://github.com/kubernetes/helm/issues/3269 
* "Meta: Build system changes" https://github.com/kubernetes/helm/issues/1968

Creating this issue to make an easy cross-reference.

**Comment by StefanScherer** [comment]
Yes, a perfect example how to do multi-arch images. It's really worth to push this to upstream repo.

**Comment by StefanScherer** [comment]
To simplify the setup for upstream we may should look at avoiding qemu binary. It's only needed for the `RUN` instruction https://github.com/jessestuart/tiller-multiarch/blob/master/Dockerfile#L10
The certs could be retrieved from the amd64 image, the certs are platform independent then copied into the arm image, maybe with a multi-stage build `FROM alpine` + install certs + `FROM $target/alpine`.

**Comment by jessestuart** [comment]
Hey all,
Thanks for the interest! Sorry this slipped under my radar last month.

@vielmetti, @StefanScherer — definitely agree it'd be nice to see some form of this merged upstream! I see there have since been a couple attempts at introducing multi-arch support to the CI pipeline (https://github.com/kubernetes/helm/pull/4200, https://github.com/kubernetes/helm/pull/4252); but there seems to be some confusion around cross-compiling **binaries** vs. building multi-arch **images** along with the appropriate Docker manifests.

@StefanScherer — love it, I hadn't thought of that. I've updated `master` with your suggestion (see #2; lots of misc changes, skip to around https://github.com/jessestuart/tiller-multiarch/pull/2/files#diff-3254677a7917c6c01f55212f86c57fbfR3 for the `Dockerfile` changes). Tested & verified on both `armv7` and `arm64` devices.

Is this still worth pursuing? If so, I'd be happy to take a stab at integrating these changes upstream. [That being said, I know there are significant changes planned for the 3.0 release; so I don't want to duplicate effort if work is already underway.]

**Comment by vielmetti** [comment]
Thanks @jessestuart !

I do think this is still worth pursuing to the degree that it dovetails with a new CI and build process for the next iteration of Helm. I don't know a timeline yet for the disappearance of Tiller, but engaging with that process to demonstrate how it should work will be a solid contribution.

**Comment by Letme** [comment]
I would like to ping about this request. Tiller image at end of 2019 is still not multi-arch and that is clear indicator that nobody is in fact working on it. The 3.0 release is also not official so while compiling helm for RPI is easy, maintaining docker images for core component like tiller should be just out of the box (imagine mixed clusters...).
